app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java

```diff
@@ -0,0 +1,221 @@
+        if (getIntent() != null && categoryName != null) {
+            Bundle arguments = new Bundle();
+            arguments.putString("categoryName", categoryName);
+            arguments.putBoolean("isParentCategory", false);
+            categoryImagesListFragment.setArguments(arguments);
+            subCategoryListFragment.setArguments(arguments);
+            Bundle parentCategoryArguments = new Bundle();
+            parentCategoryArguments.putString("categoryName", categoryName);
+            parentCategoryArguments.putBoolean("isParentCategory", true);
+            parentCategoryListFragment.setArguments(parentCategoryArguments);
+        }
+        fragmentList.add(categoryImagesListFragment);
+        titleList.add("MEDIA");
+        fragmentList.add(subCategoryListFragment);
+        titleList.add("SUBCATEGORIES");
+        fragmentList.add(parentCategoryListFragment);
+        titleList.add("PARENT CATEGORIES");
+        viewPagerAdapter.setTabData(fragmentList, titleList);
+        viewPagerAdapter.notifyDataSetChanged();
+
+    }
+
+    /**
+     * Gets the passed categoryName from the intents and displays it as the page title
+     */
+    private void setPageTitle() {
+        if (getIntent() != null && getIntent().getStringExtra("categoryName") != null) {
+            setTitle(getIntent().getStringExtra("categoryName"));
+        }
+    }
+
+    /**
+     * This method is called onClick of media inside category details (CategoryImageListFragment).
+     */
+    @Override
+    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+        tabLayout.setVisibility(View.GONE);
+        viewPager.setVisibility(View.GONE);
+        mediaContainer.setVisibility(View.VISIBLE);
+        if (mediaDetails == null || !mediaDetails.isVisible()) {
+            // set isFeaturedImage true for featured images, to include author field on media detail
+            mediaDetails = new MediaDetailPagerFragment(false, true);
+            FragmentManager supportFragmentManager = getSupportFragmentManager();
+            supportFragmentManager
+                    .beginTransaction()
+                    .replace(R.id.mediaContainer, mediaDetails)
+                    .addToBackStack(null)
+                    .commit();
+            supportFragmentManager.executePendingTransactions();
+        }
+        mediaDetails.showImage(i);
+        forceInitBackButton();
+    }
+
+
+    /**
+     * Consumers should be simply using this method to use this activity.
+     * @param context
+     * @param categoryName Name of the category for displaying its details
+     */
+    public static void startYourself(Context context, String categoryName) {
+        Intent intent = new Intent(context, CategoryDetailsActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
+        intent.putExtra("categoryName", categoryName);
+        context.startActivity(intent);
+    }
+
+    @Override
```

Comment:
There are still many methods without Javadoc it seems, such as these ones.
Please add Javadoc to all of them, thanks! :-)

Reply:
@nicolas-raoul Ok, I will search and do it for all.

Reply:
@nicolas-raoul I have added it for most of the methods now. Can you tell if we need it in any other method/class?